src/prefect/records/result_store.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+from typing import Any
+
+from prefect.exceptions import ObjectNotFound
+from prefect.results import BaseResult, PersistedResult, ResultFactory
+from prefect.utilities.asyncutils import run_coro_as_sync
+
+from .store import RecordStore
+
+
+@dataclass
+class ResultFactoryStore(RecordStore):
```

Comment:
Would we need any other type of record store? A `ResultFactory` composes a storage block, so it seems like we could write records to any storage layer with a storage block.

Reply:
Good question -- yea, right now using results for record stores gives us only "read-committed" transaction isolation, meaning two processes could be working on the same transaction key at the same time.  To support "serializable" transaction isolation, we'll need to support record stores such as redis, dynamoDB, etc.

Reply:
Would we implement that with a different class, or would we use this class with a storage block that works with Redis or DynamoDB? I'm guessing that this implementation will evolve, and I'm trying to understand how this can be extended.

Reply:
We'd need a separate class; this class is "simple" because it interacts directly with loading/saving results using the result factory, and the abstract record store only needs to interface with _serialized_ results.  E.g., a redis record store will store the output of `PersistedResult(...).json()` which might reference a result stored in S3.